### DISCUSSION TITLE
_Decimal128 unsupported in gcc?

### DISCUSSION THREAD

**Discussion by jstrodtb** [open-power-sdk/pveclib#203] [Q&A]
I am trying to use pveclib with gcc14 on Power10 and Rocky . I get this error:

vec_common_ppc.h:429:3: error: ‘_Decimal128’ does not name a type
  429 |   _Decimal128 dpd128;

This datatype should be supported in gcc, but even a toy code returns an error when I try to create this type. Any ideas what I might be doing wrong?

**Comment by munroesj52** [comment]
That is weird. This old code (back to POWER6!).

#ifndef PVECLIB_DISABLE_DFP
  /*! \brief 128 bit Decimal Float from pair of double float registers.  */
  _Decimal128 dpd128;
#endif

The code is guarded with PVECLIB_DISABLE_DFP which implies it passed the config test ./testprograms/test_ppc_DFP.c and PVECLIB_DISABLE_DFP is not defined on the compile.

Please try compiling just  ./testprograms/test_ppc_DFP.c and report back.

**Reply by munroesj52** [reply]
The gcc-14 install Ubuntu 24.04 LTS works for me on P10.

**Comment by munroesj52** [comment]
I need more info, GCC -v output, configure options used for pveclib, and configure text output?

**Reply by munroesj52** [reply]
problem solved or?
